spec/unit/filters/filter_form_builder_spec.rb

```diff
@@ -173,6 +173,21 @@ def filter(name, options = {})
         }
         body
       end
+
+      context "and a timeout error occurs" do
+        let(:body) { filter :title, as: :select, collection: ["foo"] }
+        let(:input_super_class) { Formtastic::Inputs::Base::Collections }
+        let(:db_timeout_exception) { ActiveRecord::QueryCanceled.new("ERROR: canceling statement due to statement timeout") }
+        let(:expected_exception_message) { "ERROR: canceling statement due to statement timeout while querying the values for the :title filter" }
+
+        before do
+          allow_any_instance_of(input_super_class).to receive(:collection).and_raise(db_timeout_exception)
```

Comment:
This is a hard requirement for the test to pass so it must use `expect*` instead of `allow*` as the latter can introduce false positives.

```suggestion
          expect_any_instance_of(input_super_class).to receive(:collection).and_raise(db_timeout_exception)
```